### ISSUE TITLE
BigQueryType.fromSchema annotation type mismatch

### ISSUE THREAD

**Issue by arturroo** [spotify/scio#2903]
I use Scala 2.13, Scio 0.9.0 and for tests. 
I tried:
```
    @BigQueryType.fromSchema(
        """
          |{
          | "fields": [
          |     {"name": "meta", "type": "STRING", "mode": "REPEATED", "fields": [
          |         {"name": "language"     , "type": "STRING", "mode": "NULLABLE"},
          |         {"name": "attachments"  , "type": "STRING", "mode": "REPEATED"}
          |     ]}
          | ]
          |}
          |""".stripMargin
    )
    class MySchemaClass
```
but this do not compile with error:
```
Main.scala:59:6: type mismatch;
[error]  found   : java.util.List[AnyRef]
[error]  required: ?{def asScala: ?}
[error] Note that implicit conversions are not applicable because they are ambiguous:
[error]  both method ListHasAsScala in trait AsScalaExtensions of type [A](l: java.util.List[A])jdk.CollectionConverters.ListHasAsScala[A]
[error]  and method asScalaBufferConverter in object JavaConverters of type [A](l: java.util.List[A])collection.JavaConverters.AsScala[scala.collection.mutable.Buffer[A]]
[error]  are possible conversion functions from java.util.List[AnyRef] to ?{def asScala: ?}
[error]     @BigQueryType.fromSchema(
[error]      ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
```
My build.sbt:
```
scalaVersion := "2.13.1"
scalacOptions ++= Seq(
  "-Ymacro-annotations"
)
libraryDependencies ++= Seq(
  "com.spotify" %% "scio-core" % "0.9.0",
  "com.spotify" %% "scio-bigquery" % "0.9.0",
  "com.spotify" %% "scio-test" % "0.9.0" % "test",
  "org.apache.beam" % "beam-runners-direct-java" % "2.19.0",
  "org.apache.beam" % "beam-runners-google-cloud-dataflow-java" % "2.19.0"
)
```

**Comment by nevillelyh** [comment]
`"name": "meta", "type": "STRING", "mode": "REPEATED", "fields":`
Shouldn't this be of type `RECORD`? You can't have strings inside a string.

**Comment by arturroo** [comment]
Yes, your're right! Sorry for this type error.
I've changed this to RECORD and the error is still there during compilation.
```
    @BigQueryType.fromSchema(
        """
          |{ "fields": [
          |  { "name" : "meta", "type" : "RECORD", "mode" : "REPEATED", "fields" : [
          |           { "name": "language",     "type": "STRING", "mode": "NULLABLE" }
          |         , { "name": "attachments",  "type": "STRING", "mode": "REPEATED"}
          |  ]}
          |]}
          |""".stripMargin
    )
    class MySchemaClass
```

**Comment by arturroo** [comment]
I've noticed that it can be Scala<->Java Converters.
With Scala 2.13 suggested import is ```import scala.jdk.CollectionConverters._```
Because ```import collection.JavaConverters._``` is deprecated:
https://github.com/scala/scala/blob/f7ab2748e66c3d67cd21e353471c8377f880e739/src/library/scala/collection/JavaConverters.scala#L77
```
scala.collection
@deprecated("Use `scala.jdk.CollectionConverters` instead", "2.13.0")
object JavaConverters: JavaConverters.type
 extends AnyRef
 with AsJavaConverters with AsScalaConverters
  sbt: org.scala-lang:scala-library:2.13.1:jar
```
When I change to this deprecated Import then it works. So for Scala 2.13 I believe this is an issue.
Best Regards
Artur

**Comment by regadas** [comment]
Hi @arturroo, took a look at this but was unable to replicate. I started a new project from [scio.g8](http://github.com/spotify/scio.g8) added `scio-bigquery` and added your class ... everything compiled fine.

One thing to note from your example is that `0.9.0` relies on beam `2.20.0` and you are using `2.19.0`

**Comment by regadas** [comment]
I think was able to replicate this after all 🤔

**Comment by regadas** [comment]
Closed via #2966
